app/react/Layout/DocumentsList.js

```diff
@@ -78,9 +84,16 @@ class DocumentsList extends Component {
       LoadMoreButton,
       rowListZoomLevel,
     } = this.props;
+
     let counter = (
       <span>
-        <b>{documents.get('totalRows')}</b> <Translate>documents</Translate>
+        {this.props.selectedDocuments.size > 0 && (
+          <>
+            <b> {this.props.selectedDocuments.size} </b> <Translate>selected of</Translate>
+          </>
+        )}
+        <b> {this.props.documents.get('rows').size} </b> <Translate>shown of</Translate>
+        <b> {documents.get('totalRows')} </b> <Translate>documents</Translate>
       </span>
     );
```

Comment:
This is the exact type of piece of code that should become a separate pure React component that receives a very simple couple of props.

It makes for much simpler components that have simple code and you don't get all this extra "inner components" that clutter the code.

I'm not saying we need to change this now, but it something to keep in mind.

Reply:
Totally true. In fact I think we should do it. But also (and we are proposing that in our research) adding so many small components in the current folder structure is kind of confusing. Our current modules are already stuffed of files, we might agree it's hard to look for files within one single folder. At this point I think we should balance between modularity and folder stuffing? But yeah, I think what you say is the way to go!

Reply:
As long as the folder structure is 'correct' and that the components are where they should (reusable components on shared directories and specific ones in their own module), I am OK with lots of files of components and cleaner component code.

I'm not really OK with the reverse scenario: fewer files but thousand-long files. 

Maybe this is a good topic for the tech call.

Reply:
OK, agreed. We will work that.
Also, we are pitching this topic on the next tech call as part as our research :)